platform-manageability-agent/internal/comms/comms.go

```diff
@@ -210,36 +212,95 @@ func (cli *Client) RetrieveActivationDetails(ctx context.Context, hostID string,
 	switch normalizedStatus {
 	case "not connected":
+		// Clear connecting state timer (ensures clean state for both first-time
+		// and retry activation)
+		cli.connectingStateStartTime = nil
+
 		// Execute activation command
 		rpsAddress := fmt.Sprintf("wss://%s/activate", conf.RPSAddress)
 		password := resp.ActionPassword
 		activationOutput, activationErr := cli.Executor.ExecuteAMTActivate(rpsAddress, resp.ProfileName, password)
-		ok := cli.isProvisioned(ctx, string(activationOutput), hostID)
+
+		// handles intermittent activation failures
+		// and allows the main periodic timer to retry activation in the next cycle
+		outputStr := string(activationOutput)
+		if strings.Contains(outputStr, `msg="interrupted system call"`) ||
+			strings.Contains(outputStr, "exit code: 10") {
+			log.Logger.Warnf("Interrupted system call detected for host %s - retrying next cycle", hostID)
+		}
+		if activationErr != nil {
+			log.Logger.Errorf("Failed to execute activation command for host %s: %v", hostID, activationErr)
+		}
+
+		// Check provisioning status
+		ok := cli.isProvisioned(ctx, outputStr, hostID)
 		if !ok {
 			log.Logger.Errorf("Failed to execute activation command for host %s: %v, Output: %s",
 				hostID, activationErr, string(activationOutput))
 			activationStatus = pb.ActivationStatus_ACTIVATION_FAILED
 		} else {
-			log.Logger.Debugf("Activation command output for host %s: %s", hostID, string(activationOutput))
-			// Check if activation was successful by looking for success indicators in output
+			log.Logger.Debugf("Activation command output for host %s: %s", hostID, outputStr)
 			activationStatus = pb.ActivationStatus_ACTIVATING
 			log.Logger.Debugf("setting activation status to %s: %s", activationStatus, hostID)
 		}
 	case "connecting":
-		activationStatus = pb.ActivationStatus_ACTIVATING
-		log.Logger.Debugf("setting activation status to %s: %s", activationStatus, hostID)
+		activationStatus = cli.handleConnectingState(ctx, hostID)
+
 	case "connected":
+		// Reset connecting state timestamp since we're connected
+		cli.connectingStateStartTime = nil
 		activationStatus = pb.ActivationStatus_ACTIVATED
-		log.Logger.Debugf("setting activation status to  %s: %s", activationStatus, hostID)
+		log.Logger.Debugf("setting activation status to %s: %s", activationStatus, hostID)
+
 	default:
 		log.Logger.Warnf("Unknown RAS Remote Status for host %s: %s", hostID, rasStatus)
 		activationStatus = pb.ActivationStatus_UNSPECIFIED
-		log.Logger.Debugf("setting activation status to  %s: %s", activationStatus, hostID)
+		log.Logger.Debugf("setting activation status to %s: %s", activationStatus, hostID)
 	}
 
 	return cli.reportActivationResult(ctx, hostID, activationStatus)
 }
 
+// handleConnectingState manages the connecting state with timeout and deactivation
+// triggering deactivation after 3 minutes to retry activation.
+func (cli *Client) handleConnectingState(_ context.Context, hostID string) pb.ActivationStatus {
```

Comment:
```suggestion
func handleConnectingState(cli *Client, hostID string) pb.ActivationStatus {
```
Since this does not need to use the `context` input, only the `Client`, let's pass it as the first input and not make this function part of `Client`